src/parse.cpp

```diff
@@ -248,7 +248,7 @@ expression* parser::parse_primary_expression(precedence prec) {
     source_code_span operator_span = this->peek().span();
     this->skip();
     expression* child = this->parse_expression(
-        precedence{.binary_operators = false, .commas = false});
+        precedence{.binary_operators = true, .commas = false});
```

Comment:
This change doesn't seem right. Won't it break code like this:
```js
++x + y

(++x) + y  // before your commit
++(x + y)  // after your commit
```

Reply:
> This change doesn't seem right. Won't it break code like this:

Yup it was breaking.